website/docs/opengl/c1/3-hello-texture.md

```diff
@@ -0,0 +1,447 @@
+
+### `TextureMinFilter` & `TextureMagFilter`:
+Texture `min` and `mag` filters are the filters used when the texture's final size is, respectively, less than or greater than the original size.
+For now, we will show just the main two options and in the next section (Mipmaps), some others will be shown.
+
+
+#### `Texture(Min/Mag)Filter.Linear`:
+The (bi)linear filter is the best for low-quality images. When the pixels are sampled from the texture, this filter will get the color of the nearest
+pixels of the pixel in the center of the texture coordinates and will return a linear interpolation of them.
+
+This is an example from [Learn OpenGL](https://learnopengl.com/Getting-started/Textures). See how the neighbor colors are interpolated to return a different
+color:
+
+![linear filter](../../../images/opengl/chapter1/lesson3/texParameters/filter_linear.png)
+
+
+#### `Texture(Min/Mag)Filter.Nearest`:
+The nearest filter returns the color of the center of the nearest pixel, no interpolation is done.
+
+![Nearest filter](../../../images/opengl/chapter1/lesson3/texParameters/filter_nearest.png)
+
+## Mipmaps
+Now for the last part of this tutorial. Mipmaps are an essential resource for making good renders.
+
+But first, what are Mipmaps?
+
+Mipmaps are a map of tiny versions of the texture. The following is an example of mipmap texture:
+![An example of a Mipmap texture](../../../images/opengl/chapter1/lesson3/example_mipmap.png)
+
+But what is this used for?
+
+At first look, mipmaps may seem very useless. But in reality, they are extremely important! Mipmaps are used when the texture is being rendered at a very low size,
+which can cause some weird visual artifacts, even with the use of the linear filter. The most potent visual effect is something called the "moiré effect", which can
+be tiring for the eyes, and generally look quite ugly.
+
+The purpose of a mipmap is to provide an alternative texture to the fragment shader, a one with a more appropriate size for the geometry being drawn, almost completely
+eliminating the moiré effect.
+
+An example from [Wikipedia](https://en.wikipedia.org/wiki/File:Mipmap_Aliasing_Comparison.png). It's possible to notice weird patterns
+generated far away in the render without mipmaps:
+
+![Mipmap usage example](../../../images/opengl/chapter1/lesson3/mipmap_comparation.png)
+
+But if you think that generating mipmaps by hand for all your textures is really hard work, don't worry! OpenGL provides a special method to
+do this for you.
+
+Just after setting the texture parameters, try to add this line to your code:
+```c#
+_gl.GenerateMipmap(TextureTarget.Texture2D);
+```
+
+But this is not everything, we also need to set some parameters to tell the sampler to use the mipmaps:
+
+### `GL_NEAREST_MIPMAP_NEAREST`, `GL_LINEAR_MIPMAP_NEAREST`, `GL_NEAREST_MIPMAP_LINEAR` & `GL_LINEAR_MIPMAP_LINEAR`:
+These parameters should be used as the Texture Min filter and only with textures that have a mipmap.
```

Comment:
You haven't explained what a min filter is.

Reply:
> Texture min and mag filters are the filters used when the texture's final size is, respectively, less than or greater than the original size.
but i will rewrite it

Reply:
I'm closing conversations on filtering and unifying/superseding them with [this one](https://github.com/dotnet/Silk.NET/pull/1899/files#r1477123745).